src/main/scala/scalafix/sbt/ScalafixEnable.scala

```diff
@@ -2,66 +2,184 @@ package scalafix.sbt
+      partialVersion <- CrossVersion.partialVersion(scalaV).toList
+      maybeRecommendedSemanticdbScalacV <-
+        maybeRecommendedSemanticdbScalacScalaVersion.lift(partialVersion).toList
+      scalafixResolvers0 <- (p / scalafixResolvers)
+        .get(extracted.structure.data)
+        .toList
+      semanticdbCompilerPlugin0 <- (p / semanticdbCompilerPlugin)
+        .get(extracted.structure.data)
+        .toList
+    } yield CompatibleProject(
+      p,
+      VersionNumber(scalaV),
+      semanticdbCompilerPlugin0,
+      scalafixResolvers0,
+      maybeRecommendedSemanticdbScalacV
+    )
+
+  private case class CompatibleProject(
+      ref: ProjectRef,
+      scalaVersion0: VersionNumber,
+      semanticdbCompilerPlugin0: ModuleID,
+      scalafixResolvers0: Seq[Repository],
+      maybeRecommendedSemanticdbScalacScalaV: Option[VersionNumber]
+  )
 
   lazy val command = Command.command(
     "scalafixEnable",
-    briefHelp = "Configure SemanticdbPlugin for scalafix.",
-    detail = """1. set semanticdbEnabled & semanticdbVersion
-      |2. conditionally sets scalaVersion when support is not built-in in the compiler""".stripMargin
+    briefHelp =
+      "Configure SemanticdbPlugin for scalafix on supported projects.",
+    detail = """1. set semanticdbEnabled := true
+      |2. for scala 2.x,
+      |  - set semanticdbCompilerPlugin to the scalameta version tracked by scalafix if available for scalaVersion, 
+      |  - otherwise set semanticdbCompilerPlugin to a compatible version available for scalaVersion,
+      |  - otherwise force scalaVersion to the latest version supported by the scalameta version tracked by scalafix.""".stripMargin
   ) { s =>
     val extracted = Project.extract(s)
     val scalacOptionsSettings = Seq(Compile, Test).flatMap(
       inConfig(_)(ScalafixPlugin.relaxScalacOptionsConfigSettings)
     )
     val settings = for {
-      (p, maybeFullVersion) <- collectProjects(
-        extracted,
-        maybeSemanticdbScalacFullScalaVersion
-      )
-      enableSemanticdbPlugin <- maybeFullVersion.toList.map { fullVersion =>
-        scalaVersion := fullVersion,
-      } :+ (semanticdbEnabled := true)
+      project <- collectProjects(extracted)
+      enableSemanticdbPlugin <-
+        project.maybeRecommendedSemanticdbScalacScalaV.toList
+          .flatMap { recommendedSemanticdbScalacScalaV =>
+
+            import scalafix.internal.sbt.Implicits._
+            val semanticdbScalacModule =
+              coursierapi.Dependency
+                .parse(
+                  project.semanticdbCompilerPlugin0.asCoursierCoordinates,
+                  coursierapi.ScalaVersion.of(project.scalaVersion0.toString)
+                )
+                .getModule
+            val recommendedSemanticdbV =
+              VersionNumber(BuildInfo.scalametaVersion)
+            val compatibleSemanticdbVs = Try(
+              coursierapi.Versions.create
+                .withRepositories(project.scalafixResolvers0: _*)
```

Comment:
That part is untested, so there is a risk that this fails on corporate environments. The failure fallback keeps the current behavior though, so apart from an added latency it shouldn't be a problem.

Reply:
Maybe this is out of topic in this PR but we could use the `csrConfiguration` key in sbt to configure coursier with authentication and credentials.

Reply:
I looked at that in the context of https://github.com/scalacenter/scalafix/issues/1571, but I am not sure it would help as [scalafix-core uses the Java coursier API directly](https://github.com/scalacenter/scalafix/pull/1152), not via what lm/lm-coursier.

Reply:
That's true, but it may not be so hard to translate the sbt `csrConfiguration` into the Java API with something similar to [`ToCoursier`](https://github.com/coursier/sbt-coursier/blob/master/modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala) given that both `csrConfiguration` and the coursier Java API are stable.